azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -522,23 +522,33 @@ def __init__(self, wire_client, uri, logger_):
         self.summary = []
         self._crypt_util = CryptUtil(conf.get_openssl_cmd())
 
-        try:
-            pfx_file = self._download_certificates_pfx(wire_client, uri)
-            if pfx_file is None:  # The response from the WireServer may not have any certificates
-                return
+        pfx_file = os.path.join(conf.get_lib_dir(), PFX_FILE_NAME)
+        pem_file = os.path.join(conf.get_lib_dir(), PEM_FILE_NAME)
 
-            try:
-                pem_file = self._convert_certificates_pfx_to_pem(pfx_file)
-            finally:
-                self._remove_file(pfx_file)
+        try:
+            if not self._try_download_certificates_pfx(wire_client, uri, pfx_file):
+                return  # The response from the WireServer did not have any certificates, or they were not in the expected format
```

Comment:
Shouldn't the Pem also get created in this case?

Reply:
I didn't follow corner case. If we reach this code means, daemon already started with pem file presence and then new ext_handler ensure pem file exist. So, agent restarts in any case should not make Deamon in loop.